### DISCUSSION TITLE
Delete the Teslamate

### DISCUSSION THREAD

**Discussion by yhq34** [teslamate-org/teslamate#2252] [General]
Hi Guy,

Do you guys know how to remove the current installed teslamate? I want to have a completely re-fresh install.

**Comment by cwanja** [comment]
How do you have it installed?

**Comment by yhq34** [comment]
Just followed the Advanced install with Apache2, TLS, HTTP Basic Auth

**Comment by martin0607** [comment]
docker ls
docker stop xxx(Name of above)
docker rm xxx(Name of above)

**Comment by yhq34** [comment]
thanks, How can I delete data? I reinstalled it, and found old data still exsit.

**Comment by bahree** [comment]
Here are the steps I would recommend - albeit a little extreme. Also NOTE: These would **remove _all_ the images, volumes, containers** etc. So if you have other docker containers that you want to keep you will **NOT** want to run this.

#Stop the containers:
`sudo docker-compose down
`
#Delete all volumes:
`docker volume rm $(docker volume ls -q)
`
#Delete all containers:
`docker rm -f $(docker ps -a -q)
`
#Clean up any dangling branches
`docker system prune
`
#Remove all images from the local cache
`docker rmi $(docker images -a -q)
`

**Comment by yhq34** [comment]
thanks a lot
